demo

```diff
@@ -15,160 +35,160 @@ docker-compose, if you want to run the demo, run:
+            break
+        elif [[ "$1" = "up" ]]; then
+            RUN_UP=1
+            # Not a flag, so we should break out of the loop.
+            break
+        else
+            break
+        fi
+        shift
+    done
 
-EXTRA_FILES=
+    local EXTRA_FILES_NEXT=0
+    for i in "$@"; do
+        case $i in
+            -f=*|--file=*)
+                EXTRA_FILES="$EXTRA_FILES $i"
+                ;;
+            -f|--file)
+                EXTRA_FILES="$EXTRA_FILES $i"
+                EXTRA_FILES_NEXT=1
+                ;;
+            *)
+                if [[ $EXTRA_FILES_NEXT -eq 1 ]]; then
+                    EXTRA_FILES="$EXTRA_FILES $i"
+                    EXTRA_FILES_NEXT=0
+                fi
+                ;;
+        esac
+    done
 
-CLIENT_FILES="-f docker-compose.client.yml -f docker-compose.client.demo.yml"
-ENTERPRISE_FILES="-f docker-compose.enterprise.yml"
+    ARGS=($@)
+}
 
-CLIENT=0
-ENTERPRISE=0
-RUN_UP=0
+check_tools() {
+    # The demo environment has some external dependencies upon: curl, jq
+    hash curl 2>/dev/null || { echo >&2 "The demo script requires the 'curl' tool to be available. Aborting."; exit 1; }
+    hash jq 2>/dev/null || { echo >&2 "The demo script requires the 'jq' tool to be available. Aborting."; exit 1; }
+}
 
-# The demo environment has some external dependencies upon: curl, jq
-hash curl 2>/dev/null || { echo >&2 "The demo script requires the 'curl' tool to be available. Aborting."; exit 1; }
-hash jq 2>/dev/null || { echo >&2 "The demo script requires the 'jq' tool to be available. Aborting."; exit 1; }
-
-while [[ -n "$1" ]]; do
-    if [[ "$1" = "--no-client" ]]; then
-        echo "--no-client argument is deprecated. Client is now disabled by default and can be enabled with --client"
-    elif [[ "$1" = "--client" ]]; then
-        CLIENT=1
-        echo "-- enabling client container"
-    elif [[ "$1" = "-p" ]] || [[ "$1" = "--project-name" ]]; then
-        shift
-        DOCKER_COMPOSE_PROJECT_NAME="$1"
-    elif [[ "$1" = "--kvm" ]]; then
-        echo "--kvm argument is deprecated. KVM will be enabled automatically if available"
-    elif [[ "$1" = "--enterprise-testing" ]]; then
-        # Undocumented flag, we use this for internal testing.
-        EXTRA_FILES="$EXTRA_FILES $ENTERPRISE_FILES"
-        ENTERPRISE=1
-    elif [[ "$1" = "down" ]] || [[ "$1" = "rm" ]] || [[ "$1" = "stop" ]]; then
-        # If the argument is either "down" or "rm", enable the client so that it
-        # gets cleaned up, no matter if `--client` is passed or not.
-        CLIENT=1
-        # Not a flag, so we should break out of the loop.
-        break
-    elif [[ "$1" = "up" ]]; then
-        RUN_UP=1
-        # Not a flag, so we should break out of the loop.
-        break
-    else
-        break
+enterprise_client_early_handling() {
+    if [[ $CLIENT -eq 1 ]]; then
+        if [[ $ENTERPRISE -ne 1 ]] || [[ $RUN_UP -ne 1 ]]; then
```

Comment:
Suggestion:
I guess this might as well be an explicit zero comparison?

Reply:
Yes, good point, this is just confusing.